support-workers/src/main/scala/com/gu/support/workers/lambdas/PreparePaymentMethodForReuse.scala

```diff
@@ -38,19 +37,25 @@ class PreparePaymentMethodForReuse(servicesProvider: ServiceProvider = ServicePr
       crmId <- getOrFailWithMessage(account.CrmId, s"Zuora account $accountId has not CrmId")
       paymentMethod <- toPaymentMethod(getPaymentMethodResponse, services.goCardlessService, account.PaymentGateway)
       sfContact = SalesforceContactRecord(sfContactId, crmId)
+      contribution <- Future.fromTry(state.product match {
+        case c: Contribution => Success(c)
+        case _ => Failure(new RuntimeException("not yet supported reuse payment method for other than contributions"))
```

Comment:
reuse will fail by default for non contributions - before it would have appeared to have gone ahead (but not filled in various product specific fields eg delivery date/promo code/gift contact.)